presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/json/format/UnimplementedJsonDateTimeFormatter.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.encoder.json.format;
+
+import io.prestosql.spi.type.SqlDate;
+import io.prestosql.spi.type.SqlTime;
+import io.prestosql.spi.type.SqlTimeWithTimeZone;
+import io.prestosql.spi.type.SqlTimestamp;
+import io.prestosql.spi.type.SqlTimestampWithTimeZone;
+
+public class UnimplementedJsonDateTimeFormatter
+        implements JsonDateTimeFormatter
+{
+    public String formatDate(SqlDate value)
```

Comment:
you can make methods defined here default method implementations in `JsonDateTimeFormatter` interface.
And then drop `UnimplementedJsonDateTimeFormatter` from inheritance hierarchy for concrete formatters.